cloud/services/domains.go

```diff
@@ -79,8 +79,11 @@ func EnsureAkamaiDNSEntries(ctx context.Context, mscope *scope.MachineScope, ope
 	linodeCluster := mscope.LinodeCluster
 	linodeClusterNetworkSpec := linodeCluster.Spec.Network
 	rootDomain := linodeClusterNetworkSpec.DNSRootDomain
-	fqdn := linodeCluster.Name + "-" + linodeClusterNetworkSpec.DNSUniqueIdentifier + "." + rootDomain
 	akaDNSClient := mscope.AkamaiDomainsClient
+	fqdn := linodeCluster.Name + "-" + linodeClusterNetworkSpec.DNSUniqueIdentifier + "." + rootDomain
+	if linodeClusterNetworkSpec.DNSEndpointOverride != "" {
+		fqdn = linodeClusterNetworkSpec.DNSEndpointOverride
```

Comment:
Should this contain the rootDomain?

Reply:
post our discussion, i have updated the override field to only override the subdomain